Here is an example stack trace that one of our customers ran into:
```
panic: interface conversion: coldata.column is []int32, not []int64
goroutine 1985968134 [running]:
github.com/cockroachdb/cockroach/pkg/col/coldata.(*memColumn).Int64(0xc047add700, 0x0, 0x4ae34e0, 0xc047add700)
	/go/src/github.com/cockroachdb/cockroach/pkg/col/coldata/vec.go:185 +0x74
github.com/cockroachdb/cockroach/pkg/col/colserde.(*ArrowBatchConverter).BatchToArrow(0xc047fdf080, 0x4ab8de0, 0xc047add640, 0x19d698f, 0x100000003b0a5c0, 0x0, 0x60, 0xc03201d6e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/col/colserde/arrowbatchconverter.go:170 +0x8f6
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).sendBatches(0xc01500bb90, 0x4a5e5c0, 0xc018b2e840, 0x7fa7da593fd0, 0xc02624a280, 0xc0176ac510, 0xc01500bb90, 0xc03201d6e0, 0x769c05)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:233 +0x134
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).runWithStream(0xc01500bb90, 0x4a5e5c0, 0xc018b2e840, 0x7fa7da593fd0, 0xc02624a280, 0xc0176ac510)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:304 +0x105
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).Run(0xc01500bb90, 0x4a5e500, 0xc047d197c0, 0x49e5840, 0xc000d48e10, 0xda5983eb00000001, 0x1052beae8c496021, 0xc0ccf19bfd, 0x2, 0xc0176ac510)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:164 +0x596
```
They are running on 19.2.2, and the underlying problem has been fixed in 19.2.5 (or 19.2.6), but I think it is still helpful to have this catcher.

--- a/pkg/sql/colflow/colrpc/outbox.go
+++ b/pkg/sql/colflow/colrpc/outbox.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/logtags"
 	"google.golang.org/grpc"
 )
@@ -243,6 +244,16 @@ func (o *Outbox) sendBatches(
 		}
 		o.batch = o.Input().Next(o.runnerCtx)
 	}
+	serializeBatch := func() {
+		o.scratch.buf.Reset()
+		d, err := o.converter.BatchToArrow(o.batch)
+		if err != nil {
+			colexecerror.InternalError(errors.Wrap(err, "Outbox BatchToArrow data serialization error"))
+		}
+		if _, _, err := o.serializer.Serialize(o.scratch.buf, d); err != nil {
+			colexecerror.InternalError(errors.Wrap(err, "Outbox Serialize data error"))
+		}
+	}
 	for {
 		if atomic.LoadUint32(&o.draining) == 1 {
 			return true, nil
@@ -258,14 +269,8 @@ func (o *Outbox) sendBatches(
 			return true, nil
 		}
 
-		o.scratch.buf.Reset()
-		d, err := o.converter.BatchToArrow(o.batch)
-		if err != nil {
-			log.Errorf(ctx, "Outbox BatchToArrow data serialization error: %+v", err)
-			return false, err
-		}
-		if _, _, err := o.serializer.Serialize(o.scratch.buf, d); err != nil {
-			log.Errorf(ctx, "Outbox Serialize data error: %+v", err)
+		if err := colexecerror.CatchVectorizedRuntimeError(serializeBatch); err != nil {
+			log.Errorf(ctx, "%+v", err)
 			return false, err
 		}
 		o.scratch.msg.Data.RawBytes = o.scratch.buf.Bytes()

